### ISSUE TITLE
BUG? Linq-Queries are not working after update to EFCore

### ISSUE THREAD

**Issue by sanalbakis** [dotnet/efcore#5968]
Dear folks, 

I am not sure if there is an error, but I have to change many queries after update to efcore to get the right result from database. Maybe you can help/answer me. if this is a bug or it wont work anymore like before?

You can find example query below: 

**Updated**

I got more time to check the issue, but some points I still dont understand yet!

The user was actually not "left outer join" method.

**before**: join us in UnitOfWork.Context.User on logs.UserId equals us.BaseObjectId

**after**: join us in UnitOfWork.Context.User on logs.UserId equals us.BaseObjectId into User
                from us in User.DefaultIfEmpty()

the reason that I made it:
     I could get the list of ItemLogs (at least 2 items, (before - after). After update it to EFCore, the result count of itemlogs is only 1. it gets the first item. 

the userId is required and there is already an userid in db. 

```
    public Logs GetLog(long baseObjectId)
    {
        var q = (from bo in UnitOfWork.Context.BaseObject
            join logs in UnitOfWork.Context.Logs on bo.BaseObjectId equals logs.BaseObjectId

            join us in UnitOfWork.Context.User on logs.UserId equals us.BaseObjectId into User
            from us in User.DefaultIfEmpty()

            join itlogs in UnitOfWork.Context.ItemLogs on logs.BaseObjectId equals itlogs.LogsId into ItemLogs
            from itlogs in ItemLogs.DefaultIfEmpty()

            where logs.BaseObjectId == baseObjectId
            select new Logs()
            {
                ItemLogs = ItemLogs.ToList(),
                BaseObject = bo,
                BaseObjectId = bo.BaseObjectId,
                User = us,
                UserId = us.BaseObjectId,
                ArticleNumber = logs.ArticleNumber,
                IsActivated = logs.IsActivated,
                IsDeleted = logs.IsDeleted,
                IsLocked = logs.IsLocked,
                IsPriceChanged = logs.IsPriceChanged,
                Name = logs.Name,
                ParentId = logs.ParentId.Value,

            }).FirstOrDefault();

        return q;
    }
```

another query:

in this case, the first join item-count (LanguageData - PimLng) is now always 1, instead of there are more items. I have just checked the SQL-Query, the result-count for lng-data is 2. 

```
    public ProductInformationManagement GetProductInformationManagement(long baseObjectId)
    {
        var q = (from bo in UnitOfWork.Context.BaseObject
                 join pim in UnitOfWork.Context.ProductInformationManagement on bo.BaseObjectId equals pim.BaseObjectId

                 join pimlng in UnitOfWork.Context.ProductInformationManagementLng on pim.BaseObjectId equals pimlng.ParentId.Value into PimLng
                 from pimlng in PimLng.DefaultIfEmpty()

                 join ifeature in UnitOfWork.Context.ItemFeature on pim.BaseObjectId equals ifeature.ParentId.Value into ItemF
                 from ifeature in ItemF.DefaultIfEmpty()

                 join iImage in UnitOfWork.Context.ItemImage on pim.BaseObjectId equals iImage.ParentId.Value into ItemI
                 from iImage in ItemI.DefaultIfEmpty()

                 join iFile in UnitOfWork.Context.ItemFile on pim.BaseObjectId equals iFile.ParentId.Value into ItemFile
                 from iFile in ItemFile.DefaultIfEmpty()

                 join iArtShip in UnitOfWork.Context.ItemArticleShipping on pim.BaseObjectId equals iArtShip.ProductInformationManagementId into IArtShip
                 from iArtShip in IArtShip.DefaultIfEmpty()

                 join itemArt in UnitOfWork.Context.Article on pim.BaseObjectId equals itemArt.ParentId.Value into Articles
                 from itemArt in Articles.DefaultIfEmpty()

                 where pim.BaseObjectId == baseObjectId

                 select new ProductInformationManagement()
                 {
                     BaseObjectId = pim.BaseObjectId,
                     Name = pim.Name,
                     ParentId = pim.ParentId,
                     BaseObject = bo,
                     IsLocked = pim.IsLocked,
                     Path = pim.Path,
                     Article = Articles.ToList(),
                     DirectoryId = pim.ParentId.Value,
                     CatalogPage = pim.CatalogPage,
                     ContentUnit = pim.ContentUnit,
                     ManufacturerName = pim.ManufacturerName,
                     ManufacturerTypeDescr = pim.ManufacturerTypeDescr,
                     MinOrderUnit = pim.MinOrderUnit,
                     OrderUnit = pim.OrderUnit,
                     PriceUnit = pim.PriceUnit,
                     SaleUnit = pim.SaleUnit,
                     DeliveryTimeInDay = pim.DeliveryTimeInDay,
                     ItemFeature = ItemF.ToList(),
                     ItemImage = ItemI.ToList(),
                     LanguageData = PimLng.ToList(),
                     ItemArticleShipping = IArtShip.ToList(),
                     ItemFile = ItemFile.ToList(),

                 }).FirstOrDefault();
        return q;
    }
```

Thank you. 
### Further technical details

EF Core version: 1.0.0
Operating system: Windows 10 x64
Visual Studio version: VS2015

**Comment by rowanmiller** [comment]
**EF Team Triage:** This issue is lacking enough information for us to be able to effectively triage it. In particular, it is missing the following information requested in the new issue template. Can you please provide this information?

> ### Steps to reproduce
Ideally include a complete code listing that we can run to reproduce the issue.
Alternatively, you can provide a project/solution that we can run.

BTW we're not just doing this to be mean 😄... we get a lot traffic on this project and it takes time to attempt to reproduce an issue based on fragments of information. In addition, our attempt is often unsuccessful as the exact conditions required to hit the issue are often not explicitly included in the code provided. To ensure we maximize the time we have to work on fixing bug, implementing new features, etc. we ask that folks give us a self contained way to reproduce an issue.

For a guide on submitting good bug reports, read_ [Painless Bug Tracking](http://www.joelonsoftware.com/articles/fog0000000029.html).

**Comment by sanalbakis** [comment]
Hi, sorry, I was on holiday, I can now write back.

Example models:

     public class BaseObject {
          public long Id { get; set; }

         public long BaseObjectId { get; set; } 

         public long? ParentId { get; set; }
         private BaseObject _parent;

         public virtual BaseObject Parent
         {
             get { return _parent; }
             set
             {
                 _parent = value;
                 ParentId = value != null ? value.BaseObjectId : (long?)null;
             }
         }
     }

     public class Logs {
         public long Id { get; set; }
         public long BaseObjectId { get; set; }
         public BaseObject BaseObject {get; set;}
         public string ArticleNumber { get; set; }
         public bool IsPriceChanged { get; set; }
         public bool IsLocked { get; set; }
         public bool IsActivated { get; set; }
         public bool IsDeleted { get; set; }
         
         public long UserId { get; set; }
         public User User { get; set; }
         
         public ICollection<ItemLogs> ItemLogs { get; set; }
     }

     public class ItemLogs {
          public long Id { get; set; }
          public long BaseObjectId { get; set; }
          public BaseObject BaseObject {get; set;}

          public decimal Price { get; set; }
          public decimal PriceUnit { get; set; }  
          public string Currency { get; set; }
          public string Ean { get; set; }
          public decimal Quantity { get; set; }

          public long LogsId { get; set; }
          public Logs Logs { get; set; }
     }

     public class User {
          public long Id { get; set; }
          public long BaseObjectId { get; set; }
          public BaseObject BaseObject {get; set;}

          public string Username {get; set; }     
          public string Email {get; set; }
          ....
     }

The query is to get the ItemLogs is already given on top. (There are often 2 ItemLogs-Row in the database, because of to show before-update changes.).

**Comment by rowanmiller** [comment]
@sanalbakis can you share a complete code listing for a console app, or add a project. There is a lot going on in the queries you supplied and the entities you supplied only cover some of the types being used. It would take us a while to try and recreate the entire model, and I'm not sure we'd get the same results as there is a lot we'd be guessing at.

**Comment by rowanmiller** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.
